sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.service.ts

```diff
@@ -108,4 +134,152 @@ export class EducationProgramOfferingControllerService {
       description: getOfferingNameAndPeriod(offering),
     }));
   }
+
+  /**
+   * Creates the offering model to be validated and saved using the DTO received.
+   * @param institutionId institution id.
+   * @param locationId location id.
+   **The authorization to validated if the location belongs to the institution
+   **is not expected to be performed by this method.
+   * @param programId program id.
+   * @param payload information to generate the model to perform the offering
+   * validation and persistence.
+   * @returns offering model to be validated and saved.
+   */
+  async mapOfferingAPIInDTOTOSaveOfferingModel(
+    institutionId: number,
+    locationId: number,
+    programId: number,
+    payload: EducationProgramOfferingAPIInDTO,
+  ): Promise<SaveOfferingModel> {
+    // Program information required to perform the offering validations.
+    const program = await this.programService.getEducationProgramDetails(
+      programId,
+      institutionId,
+    );
+    if (!program) {
+      throw new NotFoundException(
+        "Program to create the offering not found for the institution.",
+      );
+    }
+    return {
+      ...payload,
+      locationId,
+      studyBreaks: payload.breaksAndWeeks?.studyBreaks,
+      programContext: program,
+    };
+  }
+
+  /**
+   * Verify if all CSV validations were performed with success and throw
+   * a BadRequestException in case of some failure.
+   * @param csvValidations validations to be verified.
+   */
+  assertCSVValidationsAreValid(csvValidations: OfferingCSVValidationResult[]) {
+    const csvValidationsErrors = csvValidations.filter(
+      (csvValidation) => csvValidation.errors.length,
+    );
+    if (csvValidationsErrors.length) {
```

Comment:
If the length is not compared with a value greater than 0 the block of code in the if statement will always be executed.

Reply:
The assumption is not quite accurate.
There are a few falsy values on javascript and 0(zero) is one of them: undefined, null, NaN, 0, "" (empty string), and false.
![image](https://user-images.githubusercontent.com/61259237/189416536-c2f92481-2bcb-41dd-9b8d-be471c6dd651.png)

Reply:
Correct. I thought or assumed an empty array is also an object which would have failed the condition. Thanks for testing and clarifying, I will resolve.